doc/index.rst

```diff
@@ -1,12 +1,78 @@
 .. _framework:
 
 =====================
-Core functionalities
+ROS2 Control API
 =====================
 
 `GitHub Repository <https://github.com/ros-controls/ros2_control>`_
 
-Features
-=========
+Controller Interface
+====================
+.. doxygenfile:: controller_interface.hpp
+.. doxygenfile:: controller_state_names.hpp
+.. doxygenfile:: helpers.hpp
+.. doxygenfile:: controller_interface/visibility_control.h
 
-- :ref:`Command Line Interface (CLI) <ros2controlcli_userdoc>`
+Semantic Components
+*******************
+.. doxygenfile:: force_torque_sensor.hpp
+.. doxygenfile:: imu_sensor.hpp
+.. doxygenfile:: semantic_component_interface.hpp
+
+Controller Manager
+==================
+.. won't build with hpp
```

Comment:
I don't understand this. Why wouldn't it build with `.hpp`?

Reply:
Yeah, I should document that better.

I totally agree that the header file is preferred, that's why the majority are header files, but for some reason two files in `controller_manager` wouldn't build so I switched them to the source files as a temporary measure (it does provide some documentation, though not much). When run with the header files, `controller_manager.hpp` and `controller_spec.hpp` both produce the following output:

```
Exception occurred:
  File "/usr/local/lib/python3.8/dist-packages/breathe/renderer/sphinxrenderer.py", line 302, in create
    cls, name = DomainDirectiveFactory.python_classes[args[0]]  # type: ignore
KeyError: 'struct'
```

I briefly googled around but couldn't find anything super related. I can spend more time on it now that I've gotten a minimal example working.